### PR TITLE
fix(compiler): remove redundant annotations (followup to #1988)

### LINKED ISSUE #1872: unused resource breaks the code
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

The following code
```ts (wing)
bring cloud;

resource DeclareAndNotUse {
  c: cloud.Counter;
  init() {
    this.c = new cloud.Counter();
  }
  inflight hello(): str {
    return "hello world";
  }
}

let a = new DeclareAndNotUse();
new cloud.Function(inflight () => {
  print(a.hello());
}) as "test";
```

```sh
wing test a.w
```

### I expected this:

to compile and work

### Instead, this happened:

```sh
wingly % wing test aaa.w
fail ┌ aaa.w » root/test
     │ Error: Missing environment variable: COUNTER_HANDLE_3356e028
     │     at evalmachine.<anonymous>:27:17
     │     at exports.handler (evalmachine.<anonymous>:30:9)
     │     at evalmachine.<anonymous>:33:9
     │     at Script.runInContext (node:vm:141:12)
     │     at Object.runInContext (node:vm:297:6)
     │     at /Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/target-sim/function.inflight.js:127:12
     │     at new Promise (<anonymous>)
     │     at runInSandbox (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/target-sim/function.inflight.js:108:12)
     │     at Object.activity (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/target-sim/function.inflight.js:55:24)
     │     at Object.withTrace (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/testing/simulator.js:82:50)
     │     at Function.invoke (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/target-sim/function.inflight.js:52:29)
     │     at Simulator.runTest (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/testing/simulator.js:270:22)
     │     at async Simulator.runAllTests (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/testing/simulator.js:245:26)
     │     at async testOne (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/dist/commands/test.js:18:21)
     └     at async test (/Users/eyalkeren/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/dist/commands/test.js:10:9)
```

### Component:

Compiler

### Wing Version:

0.6.0

### Node.js Version:

_No response_

### Platform(s):

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yoav-steinberg**
In #1988 we add all the resource fields to all inflight method annotations in the resource, but all we really want is to make sure someone annotates these fields in at least 1 method (to resolve #1872).

In this PR I create a dummy `$init` inflight method which includes the required annotations so we can safely assume the host can instantiate a client to the resource with access to all inner fields.

I think this fix improves on the existing code because:
* There aren't any redundant annotations.
* We're clear that these annotations are specific for initializing the resource client, and not because of some explicit inflight field access.
* This resolves a bug in the existing code where it didn't filter the annotated fields based on whether they were capturable, reassingable, etc. We now rely on the existing code in `get_captures`.
* I was thinking that eventually we'll support custom inflight `init` methods for initializing a resource's client, these methods will also need annotations which can be added to the new `$init` annotation.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/core/resource.ts [hunk 1]
```diff
@@ -390,6 +394,9 @@ export abstract class Resource extends Construct implements IResource {
   }
 }
 
+// The `init` op is a placeholder for any annotations needed for an instnace of a resource's client to be instantiated
```
Comment (hasanaburayyan) [inline libs/wingsdk/src/core/resource.ts:397]:
```suggestion
// The `init` op is a placeholder for any annotations needed for an instance of a resource's client to be instantiated
```

**Review by hasanaburayyan — COMMENTED** [review]
This looks great and makes sense to me, Ill let @Chriscbr take a peek incase he has additional thoughts.

**Review by MarkMcCulloh — APPROVED** [review]
Makes sense to me too, nice 👍

I'll add the `do-not-merge` label if you want to wait for Chris.

**Review by Chriscbr — APPROVED** [review]
LGTM

**Pull request merged**